### ISSUE TITLE
Add support for "non-objects" like taxonomy archives

### ISSUE THREAD

**Issue by brandwaffle** [10up/ElasticPress#2676]
**Is your enhancement related to a problem? Please describe.**

There are a variety of situations where having taxonomy/archive page results would be useful. This includes things like product attributes, brands, news category pages, etc. that are all commonly searched for. The issue is that these pages aren't actually WP objects.
**Describe the solution you'd like**

Archive results (aka Tag/archive page names) will appear within other search results if matching, and function within dashboard tools like Custom Results.

**Comment by theMikeD** [comment]
Yes please. Honestly finding it hard to believe this isn't a thing already.

**Comment by joeprongen** [comment]
Would definitely be an enhancement for us. We would like to use the AutoSuggest for product_cat, products and pages to help the user find anything relating to his search term.
The terms are already indexed by the system using the feature "terms".
Is there any way that you would describe what have to be done, maybe to create the functionality by hand? Like where to hook into the javascript request object and where to eventually alter the result object to support not only "objects / post types" but also "terms".
Any help would be helpful, we can share our solution to this board for people needing this fix, before it wil maybe eventually hit your roadmap. Thanks in advance.

**Comment by blockfer-rp** [comment]
I cant believe this is a low priority feature, Archive pages for tags and categories are many sites main anchor pages. How could we just simply ignore them in the research results.

**Comment by jonoalderson** [comment]
Would love this, for [feature parity with Relevanssi](https://www.relevanssi.com/knowledge-base/indexing-taxonomy-pages/) (which I'd like to move away from).

**Comment by skora-cpl** [comment]
As a customer, we were admittedly pretty surprised that this functionality had not been built into ElasticPress. 

We have some custom post types and custom taxonomies yet their respective landing pages (archive-custom_post_type_slug.php and taxonomy-the_custom_taxonomy_as_a_slug.php) aren't indexed in ElasticPress. 

For example, we have ([podcasts](https://cpl.org/podcasts )) structured as a custom post type with podcast series represented as a custom taxonomy. 

Yet, when you search for 'podcast' or either of our podcasts' name in our site search, the [Taxonomy Archive page](https://cpl.org/podcasts/page-count) that gives an overview of the podcast series and our post type archive page - https://cpl.org/podcast - which lists recently published posts of that post type, and lists our podcast series, does not appear at all in the search results the pages are either built from a taxonomy archive or a post type archive page

Although there's not much content in there; those are high value content that should appear high in search results when certain words (podcast, page count) are searched. 

We want to make more landing pages for additional custom post types and taxonomies but it's a situation where we have to tradeoff of having them show up in search by making those landing pages as a post type of page instead of utilizing the wordpress templating system which would offer us more flexibility.
